ranger-companion.lic

```diff
@@ -0,0 +1,80 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#ranger-companion
+=end
+
+custom_require.call(%w[common])
+
+no_pause_all
+
+class Companion
+  include DRC
+
+  def initialize
+    # TODO: Add raccoon messaging
+    case bput('whistle for companion', 'Your lips are too dry', 'wolf scrambles in', 'You whistle a merry tune', 'wolf perks', 'You whistle loudly for your companion', 'purse your lips to call')
+    when 'purse your lips to call'
+      exit
+    end
+
+    while line = get
+      waitrt?
+      # Dismiss wolf if it grows past the baby stage
+      if line =~ /^.*(young|full-grown) wolf (begins|howls|paws|looks|perks|sits|pants|stands|carefully)/
+        bput('signal companion to hunt', 'suddenly catches a scent', 'is too young to hunt', 'Your companion is not', 'You have no companion', 'snapping your fingers')
+        exit
+      end
+      # A baby wolf stands up then paces back and forth nervously.
+      # A baby wolf paces back and forth.
+      if line =~ /^.*(wolf|then) paces back and forth*/
```

Comment:
Could you capture the noun and combine the wolf and raccoon cases? Something like `A (.*) paces back and forth`